src/LedgerWallet.js

```diff
@@ -158,109 +148,139 @@ class LedgerWallet {
    * first one according to derivation path
    * @param {failableCallback} callback
    */
-  async getAccounts(callback) {
-    if (!this.isU2FSupported) {
-      callback(new Error(NOT_SUPPORTED_ERROR_MSG), null);
-      return;
-    }
-    if (this.accounts !== null) {
-      callback(null, this.accounts);
-      return;
-    }
-    const chainCode = false; // Include the chain code
-    const eth = await this.getLedgerConnection();
-    const cleanupCallback = (error, data) => {
-      this.closeLedgerConnection(eth);
-      callback(error, data);
-    };
-    eth
-      .getAddress_async(this.path, this.askForOnDeviceConfirmation, chainCode)
-      .then(result => {
-        this.accounts = [result.address.toLowerCase()];
-        cleanupCallback(null, this.accounts);
-      })
-      .catch(error => cleanupCallback(error));
+  getAccounts(callback) {
+    this.getMultipleAccounts(0, 1)
+      .then(res => callback(null, Object.values(res)))
+      .catch(err => callback(err, null));
   }
 
-  async getMultipleAccounts(
-    derivationPath,
-    startingIndex,
-    accountsNo,
-    callback
-  ) {
-    if (!this.isU2FSupported) {
-      callback(new Error(NOT_SUPPORTED_ERROR_MSG), null);
-      return;
-    }
-    const chainCode = false; // Include the chain code
-    const eth = await this.getLedgerConnection();
-    const cleanupCallback = (error, data) => {
-      this.closeLedgerConnection(eth);
-      callback(error, data);
-    };
-    const addresses = {};
-    for (let i = startingIndex; i < startingIndex + accountsNo; i += 1) {
-      const path = `${derivationPath}${i}`;
-      const address = await eth.getAddress_async(
-        path,
-        this.askForOnDeviceConfirmation,
-        chainCode
+  async getMultipleAccounts(indexOffset, accountsNo) {
+    let eth = null;
+    try {
+      if (!this.isU2FSupported) {
+        return Promise.reject(new Error(NOT_SUPPORTED_ERROR_MSG));
```

Comment:
just throw error here. it will automatically reject promise (move it out of try probably)